pkg/commands/run.go

```diff
@@ -116,19 +117,28 @@ func (r *RunCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.Bui
 }
 
 // addDefaultHOME adds the default value for HOME if it isn't already set
-func addDefaultHOME(user string, envs []string) []string {
+func addDefaultHOME(u string, envs []string) []string {
 	for _, env := range envs {
 		split := strings.SplitN(env, "=", 2)
 		if split[0] == constants.HOME {
 			return envs
 		}
 	}
+
 	// If user isn't set, set default value of HOME
-	if user == "" {
+	if u == "" {
 		return append(envs, fmt.Sprintf("%s=%s", constants.HOME, constants.DefaultHOMEValue))
 	}
+
 	// If user is set, set value of HOME to /home/${user}
-	return append(envs, fmt.Sprintf("%s=/home/%s", constants.HOME, user))
+	// If uid is saved instead of username, lookup the user and supply the username
+	userObj, err := user.LookupId(u)
+	// The user is guaranteed to exist, so if err is not nil, then the username is set in the config
+	if err == nil {
+		u = userObj.Username
+	}
+
+	return append(envs, fmt.Sprintf("%s=/home/%s", constants.HOME, u))
```

Comment:
I think that if the UID is set instead of username, the value of $HOME defaults to root `/`

I tried it out with this Dockerfile:
```
FROM ubuntu
USER 1000
CMD echo $HOME
```

and ran it:
```
$ docker run test
/
```
that's what seems like is happening. We still ignore Dockerfile_test_user_run in the integration tests, which is they didn't catch it.

I couldn't find any documentation to verify this unfortunately, it just says that HOME is determined by USER [here](https://docs.docker.com/engine/reference/run/#env-environment-variables) :disappointed:

Reply:
Yeah, I can verify that that is docker's behavior, silly as it is. Should we default to that? It seems like bad behavior since specifying username sets $HOME properly.

Reply:
Any updates on this? I think copying Docker is probably the right thing to do by default.